src/NServiceBus.Transport.SQS/Configure/SqsTransport.cs

```diff
@@ -67,10 +67,24 @@ internal set
             set
             {
                 Guard.ThrowIfNull(value);
+                AssertQueueNameGeneratorIdempotent(value);
```

Comment:
I'd move this to the transport runtime part. If the user-provided queue name generator depends on stuff available only when the endpoint is running (e.g., something in the IoC container), this will always fail even if the implementation is idempotent.